src/server/ua_subscription_datachange.c

```diff
@@ -171,6 +162,9 @@ MonitoredItem_ensureQueueSpace(UA_Server *server, UA_MonitoredItem *mon) {
                 TAILQ_INSERT_TAIL(&mon->queue, overflowNotification, listEntry);
                 TAILQ_INSERT_TAIL(&mon->subscription->notificationQueue, overflowNotification, globalEntry);
             }
+            ++mon->queueSize;
+            ++sub->notificationQueueSize;
+            ++sub->eventNotifications;
```

Comment:
Why is the queue size increased here? How is the while loop supposed to terminate?

Reply:
That is a good observation,

The queue size is increase because it represents the number of elements in the queue.
And since we add an element...

Without the increase, we get an unsigned underflow.
With the increase, we ever exit the loop.

Where is the handling of overflow events described in the standard?
So we can reference that.

Reply:
Part 4 5.12.1.5 (page 61).

"If Events are lost, an Event of the type
_EventQueueOverflowEventType_ is placed in the queue. [...] It is put into the Queue of
the _MonitoredItem_ in addition to the size of the Queue defined for this _MonitoredItem_"

Sounds like it would be best to keep count of the overflows separately.